contracts/Aavegotchi/facets/WhitelistFacet.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.1;
+
+import {LibMeta} from "../../shared/libraries/LibMeta.sol";
+import {Modifiers, Whitelist} from "../libraries/LibAppStorage.sol";
+
+contract WhitelistFacet is Modifiers {
+    event WhitelistCreated(uint256 indexed whitelistId);
+    event WhitelistUpdated(uint256 indexed whitelistId);
+
+    function createWhitelist(string calldata _name, address[] calldata _whitelistAddresses) external {
+        uint256 whitelistLength = _whitelistAddresses.length;
+        require(whitelistLength > 0, "WhitelistFacet: Whitelist length should be larger than zero");
+
+        address sender = LibMeta.msgSender();
+        uint256 whitelistId = s.whitelists.length + 1;
+        address[] memory addresses;
+        Whitelist memory whitelist = Whitelist({owner: sender, name: _name, addresses: addresses});
+        s.whitelists.push(whitelist);
+        _addAddressesToWhitelist(whitelistId, _whitelistAddresses);
+
+        emit WhitelistCreated(whitelistId);
```

Comment:
I think the name should be output in the event.
Or name can just be removed completely since it's not being used anywhere else from what I can see?

Reply:
The name can be looked up on-chain by using the getWhitelist function